alphafold3_pytorch/alphafold3.py

```diff
@@ -5406,11 +5396,14 @@ def forward(
         total_atoms = molecule_atom_lens.sum(dim = -1)
         atom_mask = lens_to_mask(total_atoms, max_len = atom_seq_len)
 
-        # handle offsets for molecule atom indices
+        # handle offsets for molecule and distogram atom indices
 
         if exists(molecule_atom_indices):
             molecule_atom_indices = molecule_atom_indices + exclusive_cumsum(molecule_atom_lens)
 
+        if exists(distogram_atom_indices):
```

Comment:
This was a key bug I believe I found, since without this change, the model would be forced to learn the distogram of only a single token

Reply:
so i actually refactored that to be taken care of within the `molecule_to_atom_fn` [here](https://github.com/lucidrains/alphafold3-pytorch/blob/main/alphafold3_pytorch/inputs.py#L1167-L1169)

the reason had something to do with modified biomolecules being one atom per token, but i'll need to review it tomorrow morning as my mind is spent today

Reply:
at some point i'll also add a validation for these indices (to make sure no duplicates as well as ascending per molecule) behind [an environment flag](https://github.com/lucidrains/alphafold3-pytorch/blob/main/alphafold3_pytorch/tensor_typing.py#L62)

Reply:
Oh, this refactoring was for `molecule_lengthed_molecule_input_to_atom_input()`, not `molecule_input_to_atom_input()`, so the `pdb_input_to_molecule_input()` path wouldn't reach this point. In other words, the offsetting will currently only be used during inference, not training. However, if you add the equivalent logic to `molecule_input_to_atom_input()`, then training will also be supported (and I can then get rid of this re-offsetting in the forward() function)

Reply:
Actually, I think it'd make more sense for me to put the equivalent logic in `pdb_input_to_molecule_input()` and then get rid of this manual re-offsetting in the forward() function

Reply:
Handled per [this commit](https://github.com/lucidrains/alphafold3-pytorch/pull/168/commits/1e1afdd78c2d815a127aaaad7cfac544f8ff0af3)

Reply:
@amorehead both of our input pathways seems to pass the hard validation test! :partying_face: